pkg/controllers/restore/controller.go

```diff
@@ -574,7 +574,7 @@ func (h *handler) restoreResource(restoreObjInfo objInfo, restoreObjData unstruc
 		if err != nil {
 			return err
 		}
-		if hasStatusSubresource {
+		if hasStatusSubresource && obj.Object["status"] != nil {
```

Comment:
What if we want to restore CRs without a status? Does this still work?

Reply:
I can test, but I imagine `hasStatusSubresource` will be `false` in this case and then block wasn't running before the fix and won't be running following the fix.

Reply:
If that's the case, then this LGTM. We should confirm first how/when that's seet.

Reply:
@nickgerace , just confirmed that all kinds of resources (ones that don't have status, ones that do have status and define it, and ones that do have status but do not define it) import correctly.
I'm not sure I understand what you mean by _"we can add your new logic to the function that sets it so that it's co-located"_ - there is really not much of new logic here. We're simply ensuring restore doesn't attempt to set status if it's not available on the resource. Normally this should not happen but we do have some resources now (due to https://github.com/rancher/rancher/issues/33684) that have empty status.

Reply:
Sounds good, I will resolve this.

Ah I can explain. I am suggesting that we may want to see if we can move the `obj.Object["status"] != nil` check to where the boolean is set. If the `status` field is nil, then we set `hasStatusSubresource` to `false.`
https://github.com/rancher/backup-restore-operator/blob/c35f3bc1ecc749cf98344f7e699bb778ead4a6d6/pkg/controllers/restore/controller.go#L505

Reply:
IMO either way is fine, but I personally lean towards how it's implemented. This flag is more like "does CRD define the status?" and it's responsibility of `restoreResource` func to restore status if it's available. But I see how there could be valid arguments for what you're suggesting. Given that you're fine with the original approach, I would prefer to keep it.

Reply:
Alright let's do that then! Fair points.